Testing phase 1 merge to master

xapnsma93 opened the pull request:
Changed the user inputs to arguments. Now properly displays spells with "/" in their title using "_". Template might need work with some spells.

xapnsma93 committed:
version 2 broken, problem with jinja2 and FileSystemLoader

xapnsma93 committed:
Version 3, seems to be working with --path, --spell, both, and no args

xapnsma93 committed:
Version 4, same as version 3, commented some prints made for debugging

slowr reviewed (changes requested):
some small changes but overall it looks good

xapnsma93 committed:
version 4 with recommendations of slowr
-fixed .gitignore to include .idea
-fixed README.md formatting
-fixed spell_template.md formatting
-removed debugging comments, created breakpoints in personal IDE
-recreated requirements.txt in order to fix binary file issue
-created def saving_result(): to avoid code duplication

xapnsma93 committed:
version 4 with recommendations of slowr
-fixed .gitignore to include .idea
-fixed README.md formatting
-fixed spell_template.md formatting
-removed debugging comments, created breakpoints in personal IDE
-recreated requirements.txt in order to fix binary file issue
-created def saving_result(): to avoid code duplication

slowr commented on README.md:
```diff
@@ -1,2 +1,12 @@
 # Obsidian-DnD-5e-Spells
 Trying to fetch DnD 5e spells using API, then formating the spells to md files for Obsidian in a certain location.
+
```
usually in the readme we add the command line arguments (can be the output of --help)

slowr commented on main.py:
```diff
@@ -19,144 +31,94 @@ def get_spell_details(spell_url):
+            elif len(matching_spells) == 1:
+                spell_details = get_spell_details(matching_spells[0]['url'])
+
+                saving_result(spell_details, spell_template, save_directory)
```
the saving_result function returns false/true but you ignore it. is that desirable?

slowr commented on main.py:
```diff
@@ -19,144 +31,94 @@ def get_spell_details(spell_url):
+def saving_result(spell_details, spell_template, save_directory):
+    if spell_details:
+        # Ensure that the save directory exists
+        if not ensure_valid_directory(save_directory):
```
the save_directory is defined once and it is static. 
this means that you do not need to make this check for every spell but only once at the start of the program itself.

slowr reviewed (commented):
some more comments

xapnsma93 committed:
Delete .idea directory

xapnsma93 committed:
version 6.a
-changed README.md file to have usage
-nothing else really

xapnsma93 committed:
version 7
-removed the true/false returns in save_spell_md_file()
-removed saving_result as it was only calling another function then printing a successful message, merged it inside save_spell_md_file
-removed the constant check for save_directory, moved it after the input of --path
-replaced every saving_result to save_spell_md_file

slowr reviewed (approved):
lgtm
